src/main/kotlin/org/rust/ide/inspections/fixes/AddTypeArguments.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.inspections.getTypeArgumentsAndDeclaration
+import org.rust.lang.core.parser.RustParserUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.RsElementTypes.COMMA
+import org.rust.lang.core.psi.RsElementTypes.LT
+import org.rust.lang.core.psi.ext.*
+import org.rust.openapiext.buildAndRunTemplate
+import org.rust.openapiext.createSmartPointer
+
+class AddTypeArguments(element: RsElement) : LocalQuickFixAndIntentionActionOnPsiElement(element) {
+    override fun getText(): String = "Add missing type arguments"
+    override fun getFamilyName() = text
+
+    override fun invoke(
+        project: Project,
+        file: PsiFile,
+        editor: Editor?,
+        startElement: PsiElement,
+        endElement: PsiElement
+    ) {
+        val element = startElement as? RsElement ?: return
+        val (typeArguments, declaration) = getTypeArgumentsAndDeclaration(element) ?: return
+
+        val requiredParameters = declaration.requiredGenericParameters
+        if (requiredParameters.isEmpty()) return
+
+        val argumentCount = typeArguments?.typeReferenceList?.size ?: 0
+        if (argumentCount >= requiredParameters.size) return
+
+        val factory = RsPsiFactory(project)
+        val missingTypes = requiredParameters.drop(argumentCount).map { it.name ?: "_" }
+
+        val replaced = if (typeArguments != null) {
+            var anchor = typeArguments.typeReferenceList.lastOrNull() ?: typeArguments.lifetimeList.lastOrNull()
+            ?: typeArguments.lt
```

Comment:
Probably, it's more readable
```suggestion
            var anchor = with(typeArguments) {
                typeReferenceList.lastOrNull() ?: lifetimeList.lastOrNull() ?: lt
            }
```